File: src/main/java/com/nugurang/graphql/query/TaskQuery.java
package com.nugurang.graphql.query;

import com.nugurang.dto.TaskDto;
import com.nugurang.service.TaskService;
import graphql.kickstart.tools.GraphQLQueryResolver;
import java.util.Optional;
import org.springframework.stereotype.Service;

@Service
public class TaskQuery implements GraphQLQueryResolver {
    private final TaskService taskService;

    public Optional<TaskDto> getTask(Long id) {
        return taskService.getTask(id).map(entity -> entity.toDto());
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public TaskQuery(final TaskService taskService) {
        this.taskService = taskService;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/graphql/query/TaskQuery.kt
package com.nugurang.graphql.query

import com.nugurang.dto.TaskDto
import com.nugurang.mapper.TaskMapper
import com.nugurang.service.TaskService
import graphql.kickstart.tools.GraphQLQueryResolver
import org.springframework.stereotype.Service

@Service
class TaskQuery(
    private val taskService: TaskService,
    private val taskMapper: TaskMapper
) : GraphQLQueryResolver {
    fun getTask(id: Long): TaskDto {
        return taskService.getTask(id).let(taskMapper::toDto)
    }
}